tests/test-mptcpwrap

```diff
@@ -7,6 +7,17 @@
 
 set -e
 
+# Check if we're using the upstream kernel.
+#
+# upstream:          net.mptcp.enabled
+# multipath-tcp.org: net.mptcp.mptcp_enabled
+key=`sysctl --names --pattern 'mptcp\.enabled' net.mptcp`
```

Comment:
because of the `set -e`, I think you will still exit with an error here, no?

maybe you need something like:

```
if ! $(sysctl net.mptcp.enabled &>/dev/null); then
    # skip
```

or `if [ ! -d /proc/sys/net/mptcp/enabled ]; then`.

No?

PS: I often read it is recommended to use `$(cmd)` instead of the backquotes: more visible and inception is possible. (same for `${var}` vs `$var`, also to avoid typos)

Reply:
> because of the `set -e`, I think you will still exit with an error here, no?

`sysctl` doesn't exit with an error if the pattern doesn't match.  For example, the following `sysctl` call modified for the multipath-tcp.org kernel and run the upstream kernel box exits with a `0` status:
```
$ sysctl --names --pattern 'mptcp_enabled' net.mptcp
$ echo $?
0
```
Here's the result for the upstream kernel's `net.mptcp.enabled` sysctl on the same upstream kernel box:
```
$ sysctl --names --pattern 'mptcp\.enabled' net.mptcp
net.mptcp.enabled
$ echo $?
0
```
> maybe you need something like:
> 
> ```
> if ! $(sysctl net.mptcp.enabled &>/dev/null); then
>     # skip
> ```
> 
> or `if [ ! -d /proc/sys/net/mptcp/enabled ]; then`.
> 
> No?

Nice.  I like the `/proc` check. No need to spawn a process that way, AFAIK.

> PS: I often read it is recommended to use `$(cmd)` instead of the backquotes: more visible and inception is possible. (same for `${var}` vs `$var`, also to avoid typos)

Good points.  Is this something you'd like to see in the mptcpd 0.9 release, or does pushing such changes to the next (0.10?) release work for you?

Reply:
> > because of the `set -e`, I think you will still exit with an error here, no?
> 
> `sysctl` doesn't exit with an error if the pattern doesn't match. For example, the following `sysctl` call modified for the multipath-tcp.org kernel and run the upstream kernel box exits with a `0` status

Oh OK I see, thank you for the explanation.
But then it is not enough for my case where unit tests can run old "vanilla" kernels (< v5.6): `net.mptcp` is not present. But I see that here, the goal is not to support these kernels.

> > or `if [ ! -d /proc/sys/net/mptcp/enabled ]; then`.
> > No?
> 
> Nice. I like the `/proc` check. No need to spawn a process that way, AFAIK.

Yes indeed :)

> > PS: I often read it is recommended to use `$(cmd)` instead of the backquotes: more visible and inception is possible. (same for `${var}` vs `$var`, also to avoid typos)
> 
> Good points. Is this something you'd like to see in the mptcpd 0.9 release, or does pushing such changes to the next (0.10?) release work for you?

This can wait of course. I mean, I was just sharing a recommendation I saw which makes sense to me, up to you to follow it or not for new or old code :-)